firmware/ec/Makefile

```diff
@@ -78,7 +78,11 @@ LLIBS += -Wl,--end-group
 
 OBJCOPY = $(TOOLCHAIN)/bin/arm-none-eabi-objcopy
 
-all: oc_connect1
+all: gen_schema oc_connect1
+
+gen_schema:
```

Comment:
I meant utilizing the Makefile dependencies, something like:
```
%/auto_schema.c: %/sys_schema.json
        cd ../utilities/schema; ./mk_schema.sh
```
And then up above where MAIN_OBJS is defined, can have something like:
```
OC1_SCHEMA = ./platform/oc-sdr/schema/auto_schema.o
MAIN_OBJS += $(OC1_SCHEMA)
```
In order for that to work, would have to change the SRC_FILE search to exclude the schema/ directory (and all future schema directories?) with something like
```
SRC_FILE = $(shell find . -name '*.c' ! -path './test/*' ! -path './$(OUT)*' ! -path './platform/*/schema/*')
```
Something like this would allow `make oc_connect1` command work, and will allow us in the future to more easily add different builds that use new schema, without always having to do `make all` to get the schema script to run.

Another option is to just always run the script, so add it to `oc_connect1`, not `all`. Then we would add it to all future builds that we add as well. This is a lot easier, but I like to only run scripts when necessary by using the dependencies, so would only be run when things change.
```
all: oc_connect1
oc_connect1: gen_schema $(OUT)/OpenCellular.bin
```

Reply:
I think the second option is better, because the script does error checking and decides if it's necessary to call the Python script.  The script has more robust error checks.  I will update the makefiles.